src/Microsoft.AspNet.Localization/RequestLocalizationMiddleware.cs

```diff
@@ -142,6 +156,39 @@ private CultureInfo GetCultureInfo(IList<string> cultures, IList<CultureInfo> su
                 }
             }
 
+            if (fallbackToAncestorCulture & currentDepth < MaxCultureFallbackDepth)
+            {
+                // Walk backwards through the culture list and remove any root cultures (those with no parent)
+                for (var i = cultureNames.Count - 1; i >= 0; i--)
+                {
+                    var cultureName = cultureNames[i];
+                    if (cultureName != null)
+                    {
+                        var lastIndexOfHyphen = cultureName.LastIndexOf('-');
+                        if (lastIndexOfHyphen > 0)
+                        {
+                            // Trim the trailing section from the culture name, e.g. "fr-FR" becomes "fr"
+                            cultureNames[i] = cultureName.Substring(0, lastIndexOfHyphen);
+                        }
+                        else
+                        {
+                            // The culture had no sections left to trim so remove it from the list of candidates
+                            cultureNames.RemoveAt(i);
+                        }
+                    }
+                    else
+                    {
+                        // Culture name was null so just remove it
+                        cultureNames.RemoveAt(i);
+                    }
```

Comment:
Could collapse a few cases here and simplify the loop like:
```
var cultureName = cultureNames[i];
cultureNames.RemoveAt(i);

int index;
if (cultureName != null && (index = cultureName.LastIndexOf("-")) >= 0)
{
     cultureNames.Insert(i, cultureName.Substring(0, index);
}
```

Your call if documenting the negative cases is valuable, this is reasonable either way.